packages/webview_flutter/webview_flutter/lib/src/navigation_delegate.dart

```diff
@@ -89,23 +97,28 @@ class NavigationDelegate {
     void Function(String url)? onPageFinished,
     void Function(int progress)? onProgress,
     void Function(WebResourceError error)? onWebResourceError,
+    void Function(UrlChange change)? onUrlChange,
   }) : this.fromPlatform(
           PlatformNavigationDelegate(params),
           onNavigationRequest: onNavigationRequest,
           onPageStarted: onPageStarted,
           onPageFinished: onPageFinished,
           onProgress: onProgress,
           onWebResourceError: onWebResourceError,
+          onUrlChange: onUrlChange,
         );
 
   /// Constructs a [NavigationDelegate] from a specific platform implementation.
+  ///
+  /// {@macro webview_fluttter.navigation_delegate}
   NavigationDelegate.fromPlatform(
     this.platform, {
     this.onNavigationRequest,
     this.onPageStarted,
     this.onPageFinished,
     this.onProgress,
     this.onWebResourceError,
+    void Function(UrlChange change)? onUrlChange,
```

Comment:
This was not added as a field of the clas because there is a chance that the field can be different from the actual value. For example, a user may set this value and then call `NavigationDelegate.platform.setOnUrlChange`:

```dart
final NavigationDelegate delegate = NavigationDelegate(onUrlChange: (_) {});
delegate.platform.setUrlChange((_) {});
```

This is true for all the other callbacks, but changing them would be a breaking change.

Reply:
@mvanbeusekom suggested something similar by making the callback method setters in `PlatformNavigationDelegate` fields instead of methods. I decided against it initially because using a method allowed us to throw an `UnimplementedError` for unsupported callback methods. But we could add getters to the interface as a mix of both solutions. 

This wouldn't be possible in this PR since it is something we would have to add to the platform interface first anyways.